### PR TITLE
Document how to run examples

### PR DESCRIPTION
**Pull request by rukai**
I had assumed the triangle example wasn't functional yet, because I didn't realize I had to specify a feature. I only realized it was supposed to work when I saw a screenshot of the cube example.

I'm not sure what the future is of the triangle example? Will it remain outside of the gfx-examples folder?

### CODE REVIEW

**Comment by JohnDoneth** [comment]
+1 for this PR. I came across this same issue. I get a SEGILL (Illegal Instruction) when running without a API selected via feature.

**Comment by rukai** [comment]
I removed the `--release`.
However I just noticed the extensive docs used for gfx-hal examples https://github.com/gfx-rs/gfx#example Should I just replace my minimal attempt here with a copy of these docs.

**Comment by kvark** [comment]
That would be great, yeah! :+1:

**Comment by rukai** [comment]
I've redone the docs based on the gfx-hal docs.

I've also removed the winit feature from `examples/hello_triangle_rust`, winit is now always used.
I did this for consistency between examples and also because I don't know why you would want to run the example without winit... If i'm missing something here just let me know.

**Comment by kvark** [comment]
> I've also removed the winit feature from `examples/hello_triangle_rust`, winit is now always used.
I did this for consistency between examples and also because I don't know why you would want to run the example without winit... If i'm missing something here just let me know.

In general, there are 2 reasons one might want to not link with winit:
  1. when targeting the web
  2. when running only compute jobs

In this case, however, it's fine to always link to it. We can re-consider later.

File: README.md [hunk 1]
```diff
@@ -10,3 +10,30 @@ The implementation consists of the following parts:
   - `wgpu-bindings` - automatic generator of actual C headers
   - `wgpu-remote` - remoting layer to work with WebGPU across the process boundary
   - `wgpu-rs` - idiomatic Rust wrapper of the native library
+
+## Example
+
+To run an example, simply `cd` to the `examples` or `gfx-examples` directory, then use `cargo run` with `--features {backend}` to specify the backend (where `{backend}` is either `vulkan`, `dx12`, `dx11` or `metal`). For example:
```
[the remark this answers is not in the document]
Comment (rukai) [reply README.md]: Yeah I'll let you change this as you like, when you add the C example documentation.

File: README.md [hunk 2]
```diff
@@ -10,3 +10,30 @@ The implementation consists of the following parts:
...
+cd ../gfx-examples
+# Vulkan (Linux/Windows)
+cargo run --bin cube --features vulkan
+# Metal (macOS/iOS)
+cargo run --bin cube --features metal
+# DirectX12 (Windows)
+cargo run --bin cube --features dx12
+```
+
+These examples assume that necessary dependencies for the graphics backend are already installed. For more information about installation and usage, refer to the [Getting Started](https://github.com/gfx-rs/gfx/blob/master/info/getting_started.md) guide.
```
[the remark this answers is not in the document]
Comment (grovesNL) [reply README.md]: This is fine for now

**Pull request merged**